d3a_interface/data_classes.py

```diff
@@ -326,14 +365,19 @@ def to_json_string(self) -> str:
         return json.dumps(trade_dict, default=json_datetime_serializer)
 
     @classmethod
-    def from_json(cls, trade_string, current_time=None) -> "Trade":
+    def from_json(cls, trade_string) -> "Trade":
+        """De-serialize trade from json string."""
         trade_dict = json.loads(trade_string)
-        trade_dict["offer_bid"] = BaseBidOffer.from_json(trade_dict["offer_bid"], current_time)
-        if key_in_dict_and_not_none(trade_dict, "residual"):
-            trade_dict["residual"] = BaseBidOffer.from_json(trade_dict["residual"],
-                                                            current_time)
-        trade_dict["creation_time"] = parse(trade_dict["creation_time"])
-        if key_in_dict_and_not_none(trade_dict, "offer_bid_trade_info"):
+        trade_dict["offer_bid"] = BaseBidOffer.from_json(trade_dict["offer_bid"])
+        if trade_dict.get("residual"):
+            trade_dict["residual"] = BaseBidOffer.from_json(trade_dict["residual"])
+        if trade_dict.get("creation_time"):
```

Comment:
What are you testing here, the existence of the `creation_time` key or whether it has value or not

Reply:
I check if the key exists AND whether it is not None